tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+assert ddtrace_logger.handlers[0].maxBytes == 200000
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert "warning log" in content
+        assert re.search(LOG_PATTERN, content) is not None
+
+
+def test_debug_logs_go_to_stderr(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When setting up the default logger in debug mode without a log path,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"debug logs" in err
+    assert out == b""
+
+    code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"debug logs" in err
+    assert out == b""
+
+
+def test_debug_logs_can_go_to_file(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess, tmpdir):
+    """
+    When setting up the default ddtrace logger in debug mode,
+        it automatically logs to a file at the expected backup count.
+    """
+    env = os.environ.copy()
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "10"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 10
+assert ddtrace_logger.handlers[0].backupCount == 1
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+
+    testfiles = os.listdir(tmpdir)
+    files_created = 0
+    for file in testfiles:
+        if "testlog.log" in file:
+            files_created = files_created + 1
+
+    assert files_created == 2
```

Comment:
🤔 I wonder if it is better to collect the log file names into a list and assert an exact match, since we expect the log file names to always be consistent right?

```python
log_files = [
    filename for filename in os.listdir(tmpdir)
    if "testlog.log" in filename
]
assert log_files == [
  "testlog.log",
  "testlog.log.1",  # is this correct?
]
```

Reply:
I was looping through expecting it to be a file that started with testlog.log. I like your suggested approach, I'll update my commit for it since it does two things easily:

1. shows the list will only have 2 files created since the backup count is 1
2. easily show what the files are

Reply:
Marking this as resolved since the latest commit tests by file name.